yt_geotiff/io.py

```diff
@@ -127,6 +127,147 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
+                
+                filename=self.ds.directory+'/'+self.ds._field_filename[fname]['filename']                              
+
+                src = rasterio.open(filename, "r")
+                rasterio_window = g._get_rasterio_window(selector, src.crs, src.transform)
+
+                # round up rasterio window width and height
+                rasterio_window = rasterio_window.round_shape(op='ceil', pixel_precision=None)
+
+                # Read in the band/field
+                data = src.read(1, window=rasterio_window).astype(
+                    self._field_dtype) # could be multiband
+                data = self._transform_data(data)
+                
+                # Get resolution from load image
+                load_resolution = self.ds.resolution.d[0]
+                                
+                if src.res[0] !=load_resolution:
+                    # Calculate scale factor to adjust resolution
+                    scale_factor = src.res[0]/load_resolution
+                     
+                    # Order of spline interpolation- has to be in the range 0 (no interp.) to 5.
+                    data = self._resample(data, \
+                        fname, scale_factor, src.res[0], load_resolution, order=0) 
+                    
+                base_window = g._get_rasterio_window(selector, self.ds.parameters['crs'], self.ds.parameters['transform'])
+                rv[(ftype, fname)] = data[:int(base_window.width), :int(base_window.height)]
+
+            if self._cache_on:
+                self._cached_fields.setdefault(g.id, {})
+                self._cached_fields[g.id].update(rv)
+            return rv
+      
+        if size is None:
+            size = sum((g.count(selector) for chunk in chunks
+                        for g in chunk.objs))
+        for field in fields:
+            ftype, fname = field
+            fsize = size
+            rv[field] = np.empty(fsize, dtype="float64")
+        ind = 0
+                
+        for chunk in chunks:
+            src = None
+            for g in chunk.objs:
+                if g.filename is None:
+                    continue
+
+                gf = self._cached_fields.get(g.id, {})
+                nd = 0
+
+                for field in fields:
+                    
+                    # only for cached gridded objects
+                    if field in gf:
+
+                        # Add third dimension to numpy array
+                        for dim in range(len(gf[field].shape), 3):
+                            gf[field] = np.expand_dims(gf[field], dim)
+
+                        nd = g.select(selector, gf[field], rv[field], ind)
+
+                        self._hits += 1
+                        continue
+
+                    self._misses += 1
+                   
+                    ftype, fname = field
+                    
+                    filename=self.ds.directory+'/'+self.ds._field_filename[fname]['filename']  
```

Comment:
same here as previous related comment

Reply:
Resolved using the previous approach, i.e. changes to include the correct filepath in the _field_filename attribute.